File: javascript/basic/error_handling.mjs
import {
  PromptRails,
  NotFoundError,
  ValidationError,
  RateLimitError,
  UnauthorizedError,
  PromptRailsError,
} from "@promptrails/sdk";

const client = new PromptRails({
  apiKey: process.env.PROMPTRAILS_API_KEY,
});

try {
  await client.agents.get("nonexistent-id");
} catch (err) {
  if (err instanceof NotFoundError) {
    console.log(`Agent not found: ${err.message}`);
  } else if (err instanceof ValidationError) {
    console.log(`Invalid request: ${err.message}`);
  } else if (err instanceof RateLimitError) {
    console.log("Rate limited — retry after a delay");
  } else if (err instanceof UnauthorizedError) {
    console.log("Invalid API key");
  } else if (err instanceof PromptRailsError) {
    console.log(`API error: ${err.message}`);
  }
}

File: python/basic/error_handling.py
# /// script
# requires-python = ">=3.9"
# dependencies = ["promptrails>=0.9.0"]
# ///
#
# NOTE (API v2): promptrails 0.9.0 is not yet on PyPI. Until it publishes, run
# against the local sibling SDK from inside this folder, e.g.:
#   uv run --with-editable ../../../python-sdk main.py
# The pinned spec above reconciles automatically once 0.9.0 ships.
"""Error handling patterns."""

import os
from promptrails import PromptRails
from promptrails import (
    PromptRailsError,
    NotFoundError,
    ValidationError,
    RateLimitError,
    UnauthorizedError,
)

client = PromptRails(api_key=os.environ["PROMPTRAILS_API_KEY"])

# Handle specific errors
try:
    agent = client.agents.get("nonexistent-id")
except NotFoundError as e:
    print(f"Agent not found: {e.message}")
except ValidationError as e:
    print(f"Invalid request: {e.message}")
except RateLimitError:
    print("Rate limited — retry after a delay")
except UnauthorizedError:
    print("Invalid API key")
except PromptRailsError as e:
    print(f"API error: {e.message}")

client.close()
